src/odemis/driver/pmtctrl.py

```diff
@@ -447,6 +465,36 @@ def _sendCommand(self, cmd):
 
             return ans.rstrip()
 
+    def _tryRecover(self):
+        # no other access to the serial port should be done
+        # so _ser_access should already be acquired
+        self._recovering = True
+        self.state._set_value(HwError("PMTControl disconnected"), force_write=True)
+        self._ser_access.release() # because it will try to write on the port
+        # Retry to open the serial port (in case it was unplugged)
+        while True:
+            try:
+                self._serial.close()
+                self._serial = None
+            except Exception:
+                pass
+            try:
+                logging.debug("retrying to open port %s", self._port)
+                self._port = self._findDevice(self._portname)
+            except IOError:
+                time.sleep(2)
+            except Exception:
+                logging.exception("Unexpected error while trying to recover device")
+                raise
+            else:
+                break
```

Comment:
I think we might need it. Somehow, we need to exit the while loop once we found the right port. If we don't use else, it's just going to continue the loop forever.

Reply:
I'm not suggesting to remove the `break`, but only the `else`. It's mostly a matter of taste, but typically I think of the "else" as the special part of the code which should happen if no exception happen, even if some exceptions handler "recover" and continue further. Outside of the else is the all the code that should always happen after the code in the try/except was executed. Here, actually, both semantic fit, so it's up to you, I mostly care about adding a big comment next to the break indicating that it's a good thing to break.

Reply:
That makes sense! But then the break has to go in the try block, right? Otherwise we will break after an IOError, even though we want to continue again. Or we should re-raise the IOError if we want to use the break afterwards.
```
                try:
                    logging.debug("retrying to open port %s", self._port)
                    self._port = self._findDevice(self._portname)
                    # We found it back!
                    break
                except IOError:
                    time.sleep(2)
```